beacon-chain/powchain/service.go

```diff
@@ -616,7 +616,10 @@ func (s *Service) processBlockHeader(header *gethTypes.Header) {
 // batchRequestHeaders requests the block range specified in the arguments. Instead of requesting
 // each block in one call, it batches all requests into a single rpc call.
 func (s *Service) batchRequestHeaders(startBlock, endBlock uint64) ([]*gethTypes.Header, error) {
-	requestRange := (endBlock - startBlock) + 1
+	var requestRange uint64
+	if startBlock < endBlock {
```

Comment:
in the event it is an overflow, we should return an error here. 
so if 
```go
if start > end {
return err
}
```

Reply:
IIRC, we discussed this during the initial implementation. I wanted to return an empty range in such a case at first, but we agreed that explicitly informing the user that something wrong is happening is better than silently returning an empty range.